core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -442,6 +446,22 @@ public void setMessageCompression(boolean enabled) {
     stream.setMessageCompression(enabled);
   }
 
+  @Override
+  public Cork cork() {
+    final Cork cork = new Cork() {
+
+      @Override
+      public void close() {
+        checkState(corks.remove(this), "cork already closed");
+        if (corks.size() == 0) {
+          stream.flush();
+        }
+      }
+    };
+    checkState(corks.add(cork), "unable to add cork");
```

Comment:
Should not `checkState`. `Set.add()` returns false only if the object to be added is already in the set, which is impossible here because the `cork` is `new`ed.